### DISCUSSION TITLE
`x-default` matching the default language

### DISCUSSION THREAD

**Discussion by jorgemasta** [amannn/next-intl#582] [General]
With `next-13-rsc`, the `localePrefix` is by default `always`. This force to having URLs like:
- mysite.com/en (English version)
- mysite.com/es (Spanish version)

And the app redirects from `/` to the defined default language. E.g. from `/` to `/en`.

The problem came with the `x-default` version added in the header as alternate pages, since it doesn't include the prefix.
- `<link rel="alternate" href="https://mysite.com/en" hreflang="en" />`
- `<link rel="alternate" href="https://mysite.com/es" hreflang="es" />`
- `<link rel="alternate" href="https://mysite.com" hreflang="x-default" />`

This adds an innecesary redirect when using the unmatched language.

Should we add the default prefix to the `x-default` value?
- `<link rel="alternate" href="https://mysite.com/en" hreflang="x-default" />`

**Comment by amannn** [comment]
Note that `x-default` is used to express where a language can be picked (in this case automatically via `accept-language`). This is different from the default locale, which is used as a fallback when no other locale matches. The redirect is therefore expected.

Hope this helps!

**Reply by jorgemasta** [reply]
I'm still struggling to find the best approach for this. Ahref doesn't recommend redirects and says `x-default` should be a screen where users select their locale.

I also found this example in the [Google Developer Page](https://developers.google.com/search/docs/appearance/structured-data/intro-structured-data):
```
<link
  rel="alternate"
  hreflang="en"
  href="https://developers.google.com/search/docs/appearance/structured-data/intro-structured-data"
/>
<link
  rel="alternate"
  hreflang="x-default"
  href="https://developers.google.com/search/docs/appearance/structured-data/intro-structured-data"
/>
<link
  rel="alternate"
  hreflang="ar"
  href="https://developers.google.com/search/docs/appearance/structured-data/intro-structured-data?hl=ar"
/>
<link
  rel="alternate"
  hreflang="fr"
  href="https://developers.google.com/search/docs/appearance/structured-data/intro-structured-data?hl=fr"
/>
<link
  rel="alternate"
  hreflang="de"
  href="https://developers.google.com/search/docs/appearance/structured-data/intro-structured-data?hl=de"
/>
```

**They use the English version for the `x-default`.**
